GalaxyZooTouchTable/GalaxyZooTouchTable/Views/ExamplesPanel.xaml.cs

```diff
@@ -17,102 +11,11 @@ public partial class ExamplesPanel : UserControl
-                GeneralTransform generalTransform = SelectedElement.TransformToVisual(SelectedUIElement);
-                Point point = generalTransform.Transform(new Point());
-
-                TranslateTransform translateTransform = SelectedUIElement.RenderTransform as TranslateTransform;
-                if (translateTransform == null)
-                {
-                    translateTransform = new TranslateTransform();
-                    SelectedUIElement.RenderTransform = translateTransform;
-                }
-                translateTransform.AnimateTo(point);
-            }
-            // The item has been unselected, reset all items
-            else
-            {
-                ResetItems();
-            }
-        }
-
-        private void RemoveExtraItem()
-        {
-            foreach (var example in new ArrayList(ExampleList.SelectedItems))
-            {
-                if (example != ExampleList.SelectedItem)
-                {
-                    ExampleList.SelectedItems.Remove(example);
-                }
-            }
-        }
-
-        private void ResetItems()
-        {
-            foreach(var item in ExampleList.Items)
-            {
-                ListBoxItem element = (ListBoxItem)(ExampleList.ItemContainerGenerator.ContainerFromItem(item));
-                GeneralTransform generalTransform = SelectedElement.TransformToVisual(element);
-                Point point = generalTransform.Transform(new Point());
-
-                TranslateTransform translateTransform
-                    = element.RenderTransform as TranslateTransform;
-                if (translateTransform == null)
-                {
-                    translateTransform = new TranslateTransform();
-                    element.RenderTransform = translateTransform;
-                }
-                bool elementReachedOrigin = point.X == 0 && point.Y == 0;
-                if (elementReachedOrigin)
-                {
-                    translateTransform.AnimateTo(new Point());
-                } else
-                {
-                    AnimateToggleOpacity(element, false);
-                }
-            }
-        }
-
-        private void AnimateToggleOpacity(ListBoxItem Element, bool HideItem)
-        {
-            DoubleAnimation animation = new DoubleAnimation();
-            animation.From = HideItem ? 1.0 : 0.0;
-            animation.To = HideItem ? 0.0 : 1.0;
-            animation.Duration = new Duration(TimeSpan.FromSeconds(0.25));
-            Element.BeginAnimation(OpacityProperty, animation);
-        }
-
-        private void HideItems()
-        {
-            foreach (var example in ExampleList.Items)
-            {
-                ListBoxItem boxItem = (ListBoxItem)(ExampleList.ItemContainerGenerator.ContainerFromItem(example));
-
-                if (ExampleList.SelectedItem != example)
-                {
-                    AnimateToggleOpacity(boxItem, true);
-                }
-            }
```

Comment:
👋 bye bye code! Where is all of this logic moved to?

Reply:
All of this logic is now magically handled through the animations defined in the `Resources` section of `ExamplesPanel.xaml`

Reply:
✨ ✨